### ISSUE TITLE
[HOLD for payment 2024-11-11] [$250] mWeb/Chrome - Workspace Settings-Back arrow leads to the forward page instead of the back page

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#51107]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.50
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** N/A
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:**
**Slack conversation:**

**Issue found when executing PR** https://github.com/Expensify/App/pull/50307
## Action Performed:
Prerequisite: The user is logged in to the app
1. Go to Settings
2. Go to Workspaces
3. Go to any workspace
4. Go to Accounting
5. Tap on "Export" menu under QuickBooks Desktop connection
6. Tap on "Export date"
7. Selecting any option from the list
8. Tap the back arrow until you reach Workspaces again

## Expected Result:
The back arrow leads to the back page

## Actual Result:
The back arrow leads to the forward page instead of the back page

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [x] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

<details>
  <summary>Add any screenshot/video evidence</summary>

  
</details> 

https://github.com/user-attachments/assets/4c6561dd-7940-424a-9774-b7be5421fada

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021847870580076584100</li>
        <li>Upwork Job ID: 1847870580076584100</li>
        <li>Last Price Increase: 2024-10-20</li>
<li>Automatic offers: </li>
<ul>
<li>ikevin127 | Reviewer | 104566717</li>
<li>truph01 | Contributor | 104566719</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @jliexpensify</details>

**Comment by dylanexpensify** [comment]
Hey @jliexpensify! I'm heading out on parental leave so reassigning this! TY! 🙇‍♂️

**Comment by truph01** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The back arrow leads to the forward page instead of the back page

### What is the root cause of that problem?
- In this:

https://github.com/Expensify/App/blob/f7d08e0941b9747bf266a11264bbf47f51bafda6/src/pages/workspace/accounting/qbd/export/QuickbooksDesktopExportPage.tsx#L87

we call the goBack function with ```fallbackRoute``` leads to it navigates to new FullScreenNavigator instead of back to the existed one.
### What changes do you think we should make in order to solve the problem?
- Just need to use ```Navigation.goBack()``` in this case. 

### What alternative solutions did you explore? (Optional)

**Comment by NJ-2020** [comment]
## Proposal
### Please re-state the problem that we are trying to solve in this issue.
The back arrow leads to the forward page instead of the back page

### What is the root cause of that problem?
We pass fallback route to policy accounting page when when back button is pressed
https://github.com/Expensify/App/blob/f7d08e0941b9747bf266a11264bbf47f51bafda6/src/pages/workspace/accounting/qbd/export/QuickbooksDesktopExportPage.tsx#L87

Since we pass the `fallbackRoute` it will create a new FullScreenNavigator 
### What changes do you think we should make in order to solve the problem?
Instead of redirecting to policy account page we can pass the backTo params since we also have this page for RHP
```tsx
Navigation.goBack(route.params?.backTo);
```
https://github.com/Expensify/App/blob/66cf824036a51183d73dab2cc621901e43fe3196/src/libs/Navigation/linkingConfig/config.ts#L421

### What alternative solutions did you explore? (Optional)

**Comment by jliexpensify** [comment]
I'm on Android 51-1 (Pixel 7) and can inconsistently reproduce this. I'll set it as External in case @ikevin127 can consistently reproduce.

**Comment by truph01** [comment]
@ikevin127 Can you reproduce the bug?

**Comment by ikevin127** [comment]
Will attempt to reproduce and if I can will proceed with reviewing proposals, otherwise let you know.

**Comment by ikevin127** [comment]
I was able to reproduce consistently even though not being able to connect **Quickbooks Desktop**, but the issue is present in **Quickbooks Online** connection as well:

[android-mweb-repro.webm](https://github.com/user-attachments/assets/3be86484-fb84-4581-91e0-390a757f927f)

Given this, I Identified the following components where the same navigation go back with fallback logic is being used and by looking at previous PRs (like [this one](https://github.com/Expensify/App/pull/46511/files#diff-63a78024612aa0f2a697990124781a6e7167e60f28b5e9df1e5e3e00868ecffdR85)) I couldn't find a specific reason as to why the fallback was added.

**These are the components where the same logic is used:**

`QuickbooksImportPage`
`QuickbooksAdvancedPage`
✅ `QuickbooksExportConfigurationPage`
`QuickbooksDesktopAdvancedPage`
`QuickbooksDesktopImportPage`
✅ `QuickbooksDesktopExportPage`

`SageIntacctAdvancedPage`
`XeroOrganizationConfigurationPage`

The only ones which lead to our issue from what I tested are the ✅ ones and I'm not saying we should remove the fallback from all of them as that would be out of scope for our issue and also we would risk regressions.

**Comment by ikevin127** [comment]
Given the context from the comment above, @truph01's [proposal](https://github.com/Expensify/App/issues/51107#issuecomment-2423370557) looks good to me since it identified the root cause and proposed a working solution which fixes our issue.

🎀👀🎀 C+ reviewed

**Comment by ikevin127** [comment]
@NJ-2020 Thanks for your proposal. Unfortunately it came in second, the only difference being the replacement of the fallback route with `route.params?.backTo` which I don't see a strong reason for, but if you think I'm missing something and having no fallback would be a bad idea, please let us know why.

**Comment by FitseTLT** [comment]
@ikevin127 the purpose of `backTo` is usually to provide a fallbackRoute in the case of direct link access, especially in small screens, are you sure the proposed changes will not break that?

_FYI we have an [issue](https://github.com/Expensify/App/issues/50695#issuecomment-2412172339) with the same RCA as this one and proposing a general fix that will prevent similar issues as this 👍_

**Comment by ikevin127** [comment]
> especially in small screens, are you sure the proposed changes will not break that?

@FitseTLT Unless you tested the selected solution in this specific issue's case and have found that it breaks something - I think we're good to go here.

##

I looked over at your proposal's RCA and solution from the other issue and the solution is suggesting calling `goBack` and `navigate` one after the other which means calling 2 functions, honestly I'd be reserved to move forward with that solution because it doesn't show a deep understanding of the RCA nor an elegant solution.

##

If CME (@mountiny) considers that we should hold for the other issue / proposed solution, I won't have anything against that - as this is not about who get's to push their solution first 👍

**Comment by truph01** [comment]
@ikevin127 I saw in this [docs](https://help.expensify.com/articles/expensify-classic/connections/quickbooks-desktop/Connect-To-QuickBooks-Desktop), connecting QBD requires Windows device but I don't have. Can you connect it?

![image](https://github.com/user-attachments/assets/10ea250b-1d59-4e09-91be-12af42bf0d67)

**Comment by ikevin127** [comment]
@truph01 Wasn't able to connect QBD for the same reason but we have the same issue on Quickbooks Online integration which you should be able to connect - you can show that in your videos for the PR Checklist and make sure to apply the fix for the Quickbooks Desktop integration.

**Comment by truph01** [comment]
I am working on the PR

**Comment by ikevin127** [comment]
# BugZero Checklist:

- [ ] **[Contributor]** Classify the bug:

<details>
<summary>Bug classification</summary>


Source of bug:
  - [ ] 1a. Result of the original design (eg. a case wasn't considered)
  - [ ] 1b. Mistake during implementation
  - [ ] 1c. Backend bug
  - [ ] 1z. Other:

Where bug was reported:
  - [ ] 2a. Reported on production
  - [ ] 2b. Reported on staging (deploy blocker)
  - [ ] 2c. Reported on a PR
  - [ ] 2z. Other:

Who reported the bug:
  - [ ] 3a. Expensify user
  - [ ] 3b. Expensify employee
  - [ ] 3c. Contributor
  - [ ] 3d. QA
  - [ ] 3z. Other:

</details>

- [ ] **[Contributor]** The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake.

    Link to comment:

- [ ] **[Contributor]** If the regression was CRITICAL (e.g. interrupts a core flow) A discussion in [#expensify-open-source](https://app.slack.com/client/E047TPA624F/C01GTK53T8Q) has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner.

    Link to discussion:

- [ ] **[Contributor]** If it was decided to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps using the template below to ensure the same bug will not reach production again.

<details>
<summary>Regression Test Proposal Template</summary>


- [ ] **[BugZero Assignee]** Create a GH issue for creating/updating the regression test once above steps have been agreed upon.

    Link to issue:

## Regression Test Proposal
### Precondition:


-

### Test:


1.

Do we agree 👍 or 👎


</details>

**Comment by jliexpensify** [comment]
Payment Summary

- C: @truph01 $250
- C+: @ikevin127 $250

[Upwork job](https://www.upwork.com/jobs/~021847870580076584100)

**Comment by jliexpensify** [comment]
Paid and job closed!
